core-services/chat/src/main/java/org/silverpeas/core/chat/servers/EJabberdServer.java

```diff
@@ -0,0 +1,173 @@
+    SilverLogger.getLogger(this)
+        .info("Add relationships between {0} ({1}) and {2} ({3})", user1.getDisplayedName(),
+            user1.getId(), user2.getDisplayedName(), user2.getId());
+    ChatUser chatUser1 = ChatUser.fromUser(user1);
+    ChatUser chatUser2 = ChatUser.fromUser(user2);
+    request("add_rosteritem",
+        o -> o.put("localuser", chatUser2.getChatLogin())
+            .put("localserver", chatUser2.getChatDomain())
+            .put("user", chatUser1.getChatLogin())
+            .put("server", chatUser1.getChatDomain())
+            .put("nick", chatUser1.getDisplayedName())
+            .put("subs", "both"));
+    request("add_rosteritem",
+        o -> o.put("localuser", chatUser1.getChatLogin())
+        .put("localserver", chatUser1.getChatDomain())
+        .put("user", chatUser2.getChatLogin())
+        .put("server", chatUser2.getChatDomain())
+        .put("nick", chatUser2.getDisplayedName())
+        .put("subs", "both"));
+  }
+
+  @Override
+  public void deleteRelationShip(final User user1, final User user2) throws ChatServerException {
+    SilverLogger.getLogger(this)
+        .info("Delete relationships between {0} ({1}) and {2} ({3})", user1.getDisplayedName(),
+            user1.getId(), user2.getDisplayedName(), user2.getId());
+    ChatUser chatUser1 = ChatUser.fromUser(user1);
+    ChatUser chatUser2 = ChatUser.fromUser(user2);
+    request("delete_rosteritem",
+        o -> o.put("localuser", chatUser2.getChatLogin())
+            .put("localserver", chatUser2.getChatDomain())
+            .put("user", chatUser1.getChatLogin())
+            .put("server", chatUser1.getChatDomain()));
+    request("delete_rosteritem",
+        o -> o.put("localuser", chatUser1.getChatLogin())
+        .put("localserver", chatUser1.getChatDomain())
+        .put("user", chatUser2.getChatLogin())
+        .put("server", chatUser2.getChatDomain()));
+  }
+
+  @Override
+  public boolean isUserExisting(final User user) throws ChatServerException {
+    ChatUser chatUser = ChatUser.fromUser(user);
+    return request("check_account",
+        o -> o.put("user", chatUser.getChatLogin())
+            .put("host", chatUser.getChatDomain()),
+        r -> r.readEntity(Integer.class) == 0);
+  }
+
+  private void request(String command,
+      Function<JSONCodec.JSONObject, JSONCodec.JSONObject> argsProvider) {
+    request(command, argsProvider, r -> null);
+  }
+
+  private <T> T request(String command,
+      Function<JSONCodec.JSONObject, JSONCodec.JSONObject> argsProvider,
+      Function<Response, T> responseProcessor) {
+    Response response = null;
+    try {
+      response = this.requester.at(url, command).header("X-Admin", "true").post(argsProvider);
+
+      if (response.getStatus() != 200) {
+        logger.error("Failed to {0}: {1}", command, response.getStatusInfo().getReasonPhrase());
+        if (response.getStatus() == 403) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 403 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)